flashcard.koplugin/main.lua

```diff
@@ -0,0 +1,166 @@
+
+    self.flashcard_count = self.flashcard_count + 1
+    highlight.last_shown = os.time()
+
+    if choice == CardChoice.YES then
+        if highlight.n == 0 then
+            highlight.interval = 1
+        elseif highlight.n == 1 then
+            highlight.interval = 6
+        else
+            highlight.interval = highlight.interval * highlight.efactor
+        end
+        highlight.n = highlight.n + 1
+        highlight.efactor = highlight.efactor + 0.15
+    elseif choice == CardChoice.VAGUELY then
+        if highlight.n == 0 then
+            highlight.interval = 1
+        elseif highlight.n == 1 then
+            highlight.interval = 3
+        else
+            highlight.interval = 6
+        end
+        highlight.n = 0
+        highlight.efactor = highlight.efactor - 0.25
+    else
+        highlight.n = 0
+        highlight.interval = 1
+        highlight.efactor = highlight.efactor - 0.5
+    end
+    if highlight.efactor < 1.3 then
+        highlight.efactor = 1.3
+    end
+    self.data[highlight.time] = highlight
+
+    if self.flashcard_count >= self.flashcard_amount then
+        self.flashcard_count = 0
+        self.db:updateDB(self.data)
+    end
+end
+
+function Flashcard:displayFlashcard(highlight)
+    UIManager:show(FlashcardWidget:new{
+        title_text = _("Do you rerember this highlight/note?"),
+        text = T(_("Title: %1\r\n Author: %2\r\n\r\n%3"), highlight.title, highlight.author, highlight.text),
+        no_callback = function()
+            self:displayFlashcardCallback(highlight, CardChoice.NO)
+        end,
+        vaguley_callback = function()
+            self:displayFlashcardCallback(highlight, CardChoice.VAGUELY)
+        end,
+        yes_callback = function()
+            self:displayFlashcardCallback(highlight, CardChoice.YES)
+        end
+    })
+end
+
+function Flashcard:getHighlights()
+    local highlights = {}
+    local i = 0
+
+    for _ignore0, highlight in pairs(self.data) do
+        local days_passed = (os.time() - highlight.last_shown) / self.seconds_in_a_day
+        if days_passed >= highlight.interval then
+            highlights[i] = highlight
+            i = i + 1
+            if i >= self.flashcard_amount then
+                break
+            end
+        end
+    end
+
+    if #highlights < self.flashcard_amount then
+        self.flashcard_amount = #highlights
+    end
+
+    logger.dbg("highlights: ", highlights)
+    return highlights
+end
+
+function Flashcard:startFlashcardDisplay()
+    self.flashcard_count = 0
+    for _ignore, highlight in pairs(self:getHighlights()) do
+        self:displayFlashcard(highlight)
+    end
+end
+
+function Flashcard:init()
+    self.data = self.db:parseDB()
+
+    self.ui.menu:registerToMainMenu(self)
+end
+
+function Flashcard:addToMainMenu(menu_items)
+    menu_items.flashcard = {
+        enabled_func = function()
+            return not self.db.parsing
+        end,
+        text = _("Flashcard Trainer"),
```

Comment:
Like I said it doesn't matter too much here in contrib, but in front that should be:

```suggestion
        text = _("Flashcard trainer"),
```